src/archival/utils.rs

```diff
@@ -59,6 +79,51 @@ pub async fn is_row_exists(pool: &PgPool, row_id: i32) -> bool {
     }
 }
 
+pub async fn make_archival_network_request(
+    url: &str,
+) -> Result<String, Box<dyn std::error::Error + Send + Sync>> {
+    let endpoint_url = "https://web.archive.org/save";
+    let mut headers = header::HeaderMap::new();
+    headers.insert("Accept", "application/json".parse().unwrap());
+    headers.insert(
+        "Authorization",
+        "LOW iJN8ly6eMroQjKfd:TxLzPGdXKMWvLLuY".parse().unwrap(),
+    );
+    headers.insert(
+        "Content-Type",
+        "application/x-www-form-urlencoded".parse().unwrap(),
+    );
+
+    let client = Client::builder()
+        .redirect(reqwest::redirect::Policy::none())
+        .build()
+        .unwrap();
+
+    let response = client
+        .post(endpoint_url)
+        .headers(headers)
+        .body(format!("url={}", url))
+        .send()
+        .await?;
+    let response_status = response.status();
+    let response_text = response.text().await?;
+
+    if let Ok(result_ok) = serde_json::from_str::<ArchivalSuccessResponse>(&response_text) {
```

Comment:
When a json can be of multiple forms, it's better to deserialize it as a result. Try deserializing as `Result<ArchivalSuccessResponse, ArchivalErrorResponse>`.

Reply:
the result can also contain HTML (the case when the site is down for maintainance). Doing your way, I can only check for `ArchivalSuccessResponse` , `ArchivalErrorResponse`  (or can I?).  I wanted to log 3 different results:
- Success (json that contains job_id)
- Error (json containing status and message)
- Html response (in this case I am currently thinking to return the response status, and the whole text)

Currently I am using `.text()` on the reqwest response, and then using serde to convert it to json. 
There is another way `.json()`, but I need to pass the correct type at compile time. 
Either way, I would have to check for all 3 types.
Is there any idiomatic way to do this?

Reply:
Hi, sorry I got busy with some stuff, couldn't test the suggestions earlier.  
Have made necessary changes in the latest commit.
Thanks for the help!